compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategiesGPU.cpp

```diff
@@ -216,14 +242,12 @@ static bool matchGPUReduction(linalg::LinalgOp op,
                               GPUReductionStrategyInfos &info) {
   // TODO: match the sequence the strategy supports.
   StructuredOpMatcher pattern, fill, leadingEltwise, trailingEltwise;
-  makeReductionMatcher(pattern, fill, leadingEltwise, trailingEltwise);
+  makeReductionMatcher(pattern, fill, leadingEltwise, trailingEltwise,
+                       info.reductionDimensionSize);
   if (!matchPattern(op, pattern)) return false;
 
-  info.hasLeadingEltwise = leadingEltwise.getCaptured() != nullptr;
-  info.hasTrailingEltwise = trailingEltwise.getCaptured() != nullptr;
-
-  // Hardcoded workagroup size, this could be deduced from the reduction dim.
-  info.workgroupSize = {32, 2, 1};
+  // Hardcoded workgroup size, this could be deduced from the reduction dim.
+  info.workgroupSize = {32, 1, 1};
```

Comment:
This should connect to the decision made above (i.e. we should return numThreadsX/Y/Z)

Reply:
Though I don't like the hardcoding that was there before this patch, I don't see where this decision is being made.